### ISSUE TITLE
AI activates some activated abilities as soon as it can with no reason

### ISSUE THREAD

**Issue by MorihiroS** [Card-Forge/forge#5604]
**Describe the bug**
AI activates activated abilities of some cards as soon as it can with no reason. It's ineffectively.
Some of this cards:
- [ ] Ugin's Labyrinth (second ability)
- [ ] Canopy lands
- [ ] Zuran Orb
- [ ] Skirk Prospector

**To Reproduce**
1. AI plays Ugin's Labyrinth
2. AI uses second ability of Ugin's Labyrinth immediately

or

1. AI play Fiery Islet and casts Ragavan, Nimble Pilferer
2. AI successfully attacks with Ragavan, Nimble Pilferer and takes Treasure token
3. AI sacrifices Fiery Islet and now it hasn't lands

or

1. AI casts Zuran Orb
2. AI sacrifices all of its lands immediately

or

1. AI has one untapped land
2. AI casts Skirk Prospector for land's mana
3. AI sacrifices Skirk Prospector and casts Aether Vial

**Expected behavior**
AI should activate activated abilities of this and similar cards in accordance with the situation.
For example:
- AI activates second ability of the Ugin's Labyrinth only after it has enough mana to cast exiled card or in response to an attempt to destroy the Ugin's Labyrinth;
- AI sacrifices canopy lands only after it already has X lands and/or no cards in its hand
- I don't know, may be AI should activate the Zuran Orb ability when it has less than X life or when Titania, Protector of Argoth is on the battlefield
- AI doesn't sacrifice Skirk Prospector before it has 2+ lands and only for cast a spell with cmc = number of its lands + 1

**Additional context**
Similar with #4214

**Desktop**
 - OS: Windows 10
 - Java Version: 1.8.0_202-b08
 - Forge Version: 1.6.64-SNAPSHOT-07-13

**Screenshots**

![labir](https://github.com/user-attachments/assets/711b8311-586b-4bb0-ae08-a76120d3bf7c)
![canopy](https://github.com/Card-Forge/forge/assets/115870280/f68d5c6c-5441-42ff-9880-061c1cdbd1bf)
![canopy2](https://github.com/Card-Forge/forge/assets/115870280/f114855b-ad36-472c-92cc-f8b39dcb8769)
![zuran](https://github.com/Card-Forge/forge/assets/115870280/ae692c2f-4850-47c9-8f71-e37ab9e71c6e)
![skirk](https://github.com/Card-Forge/forge/assets/115870280/f31d89b9-0080-4478-8c64-f9bcd52e85be)

**Comment by MorihiroS** [comment]
Added a problem with Eladamri, Korvecdal

**Comment by MorihiroS** [comment]
@Hanmac, need AI label, keep label and assign someone, please, Issue contains 2 MH3 cards!

**Comment by MorihiroS** [comment]
@Agetian, is it realistic to fix at least the first two cards? The first one prevents the AI from playing a deck with eldrazi, the second one breaks the manabase of fast aggro decks

**Comment by Agetian** [comment]
Can you please clarify which two cards specifically you're referring to? I can take a look, though I can't promise to do it very quickly, sadly.

**Comment by MorihiroS** [comment]
@Agetian, no problem! The first two on the list: Ugin's Labyrinth (second ability) and Canopy lands

**Comment by MorihiroS** [comment]
Add Pyrite Spellbomb

**Comment by richardleggett** [comment]
This seems related, from DSK: Marvin, Murderous Mimic immediately becomes +3/+3 on enter because Possessed Goat has a: Pay 3, Discard a card activated ability. It activates abilities of other creates immediately without paying activation costs.

If this is unrelated please let me know I can start a new Issue.

<img width="567" alt="Screenshot 2024-09-17 at 21 22 59" src="https://github.com/user-attachments/assets/f617fb8a-bb79-45e0-b148-5c5716aa7ea0">

**Comment by MorihiroS** [comment]
> This seems related, from DSK: Marvin, Murderous Mimic immediately becomes +3/+3 on enter because Possessed Goat has a: Pay 3, Discard a card activated ability. It activates abilities of other creates immediately without paying activation costs.
> 
> If this is unrelated please let me know I can start a new Issue.

@richardleggett, if it activates abilities of other creatures immediately _without paying activation costs_, this requires a separate issue

**Comment by MorihiroS** [comment]
@Agetian, can you do something with at least Ugin's Labyrinth (second ability) and Canopy lands?

**Comment by MorihiroS** [comment]
@Agetian, can you do something with at least Ugin's Labyrinth (second ability) and Canopy lands? It's unplayable...

![burn1](https://github.com/user-attachments/assets/8019ccb2-b332-42ee-895c-2e8ca3feeea0)
![ugin1](https://github.com/user-attachments/assets/e5813867-5e8c-4b29-a8b5-bf43488df8a8)
![ugin2](https://github.com/user-attachments/assets/08bce718-b91f-4fd0-b747-1a32c3da642e)

**Comment by MorihiroS** [comment]
@Agetian, I'm pleasantly bummed, wasn't hoping for those annoying bugs to be fixed anymore! 

There is the same problem with Greater Gargadon, but I'm not sure how to find the optimal condition for AI use

**Comment by MorihiroS** [comment]
@Agetian, Add Greater Gargadon

**Comment by Hanmac** [comment]
@MorihiroS `Greater Gargadon` is part of a Combo with Balance Effects
I don't think there is a nice logic for this to be used (instead of never)

**Comment by MorihiroS** [comment]
@Hanmac I already wrote about Restore Balance in Expected behavior, but I'll add about Balance and Magus of the Balance

**Comment by MorihiroS** [comment]
@Hanmac I already wrote about Restore Balance in Expected behavior, but I'll add Balance, Balancing Act and Magus of the Balance

And I've described how this can be roughly implemented
